docs/plots/backends.rst

```diff
@@ -0,0 +1,318 @@
+a long method that is responsible for points, lines, and errorbars), but
+by "what is the input data type":
+
+- `~sherpa.plot.backend.BaseBackend.plot` (for scatter plots with markerstyle
+  set, for line plots with linestyle set, and for errorbars with ``xerr`` or
+  ``yerr`` set to `True`); accepts (x, y) data with optional error bars in each
+  dimension. Data can be scalar (for a single marker), or array-like.
+- `~sherpa.plot.backend.BaseBackend.histo` (similar to plot, but with
+  "histogram-style" lines); accepts (xlo, xhi, y) data with optional xerr, yerr.
+- `~sherpa.plot.backend.BaseBackend.contour` for (x0, x1, z) data
+- `~sherpa.plot.backend.BaseBackend.image` for (x0, x1, z) data on a regular
+  grid. An image is different from a contour in the sense that an image is
+  pixelated on a regular grid, while a contour can in principle describe a
+  continuous quantity or an irregular grid, even if the current implementation
+  may not provide that flexibility.
+
+Annotations
+-----------
+
+Backends should also implement the following annotation functions. They do not
+depend on the data plotted, but just annotate the plot, e.g. a
+`~sherpa.plot.RatioPlot` shows the ratio between data and model and can use an
+annotation to mark the ``ratio=1`` line.
+
+- `~sherpa.plot.backend.BaseBackend.hline` (horizontal across the entire axes)
+- `~sherpa.plot.backend.BaseBackend.vline` (vertical across the entire axes)
+
+Other annotations (e.g. text labels) might be added to the API in the future.
+For this reason new backends should inherit from
+`~sherpa.plot.backend.BasicBackend`. Any function added to the API will be
+implemented in `~sherpa.plot.backend.BasicBackend` as a no-op with a
+warning to the user like "Feature XYZ is not available in your backend ". That
+way, all Sherpa plots can immediately make use of newly added functions without
+breaking existing plotting backends; the worst that happens is that not all
+annotation will be visible in every backend.
+
+Return values
+-------------
+
+Sherpa does not expect a specific return argument from any plotting function,
+but they are allowed to have return values if that is helpful for their internal
+implementation, e.g. in the matplotlib backend, plotting a line might return a
+line object so that error bars plotted later can use ``line.color`` to match the
+color of that line.
+
+Creating plots and panels, clearing and overplotting
+----------------------------------------------------
+At this stage, we keep the existing API for creating plot and panels, for
+clearing and overplotting, i.e. each of the plotting functions above accepts
+ the following arguments: title, xlabel, ylabel, xlog, ylog, overplot, clearwindow
+
+Multi-panels plot can be set with clear_window, set_subplot, set_jointplot
+[copy documentation from CAIO 4.14 pylab_backend.clear_window etc. into here because that describes use and function]
+
```

Comment:
Do you include the limits on different axis when you create a plot? or the limits are somehow autogenerated?

Reply:
I would be very surprised if limits can be set here, as that would be a big change in functionality (not necessarily a bad one, just not one I'd expect here).

Also, CIAO is typo-d here.

Reply:
In this PR, I try ot keep any changes to the API as minimal as possible. Limits are created as before (i.e. whatever the plotting backend chooses, typically the next round value below and above the mean) until the user resets it, see. e.g. https://cxc.cfa.harvard.edu/sherpa/threads/setplot_manual/ for instructions. We can have a separate issue if we want to add parameters for the limits, just to confirm what @DougBurke said above: Adding new keywords for the limits is out of scope for this PR.